lib/routes/categories.js

```diff
@@ -6,37 +6,39 @@ const bodyParser = require('body-parser').json();
 router
     .post('/', bodyParser, (req, res, next) => {
         new Category(req.body)
-          .save()
-          .then(category => res.send(category))
-          .catch(next);
+            .save()
+            .then(category => res.send(category))
+            .catch(next);
     })
     .get('/', (req, res, next) => {
         Category.find()
+            .populate('expenses')
             .lean()
-            .then(categories => res.send(categories))
+            .then(categories => setTimeout(() => res.send(categories), 1000))
             .catch(next);
     })
     .get('/:id', (req, res, next) => {
         const id = req.params.id;
         Category.findById(id)
-          .lean()
-          .then(category => {
-              if (!category) throw { code: 400, error: `${id} not found` };
-              else res.send(category);
-          })
-          .catch(next);
+            .populate('expenses')
+            .lean()
+            .then(category => {
+                if (!category) throw { code: 400, error: `${id} not found` };
+                else res.send(category);
+            })
+            .catch(next);
     })
     .put('/:id', bodyParser, (req, res, next) => {
         Category.findByIdAndUpdate(req.params.id, req.body, {new: true})
```

Comment:
For updating a resource, you should be running your validators. If the form on the front-end is empty for each field and you click update, it'll update the MongoDB document. For the third argument, set `runValidators` to `true`...

```js
{
  new: true,
  runValidators: true,
}
```